Converted all the code to typescript. …

KolCrooks opened the pull request:
There are still some problems with some of the typings for the parser and the constants have duplicate keys(idk how to fix this). I also don't really know how to test any of this. I would recommend that you merge this into a new branch or clone my fork so that you can make sure that this stuff works. Look at the //TODO comments to see some stuff that I wasn't able to figure out.

KolCrooks committed:
Converted all the code to typescript. There are still some problems with some of the typings for the parser and the constants have duplicate keys(idk how to fix this)

ANovokmet commented on dist/solution.d.ts:
```diff
@@ -0,0 +1,2 @@
+export {};
```
This declaration it invalid. Solution.ts needs a default export.

ANovokmet commented on src/project.ts:
```diff
@@ -0,0 +1,113 @@
+    solution?: { path: string; };
+    xmlContent: XMLContent | undefined;
+    //TODO I use any a lot here but I really should have this as a concrete type. I just don't know what type that would be
+    itemGroup: { Reference: any[]; Compile: any[]; None: any[]; Content: any[]} = {Reference: [], Compile: [], None: [], Content: []};
```
This is good. { [key:string]: any[] } is acceptable too.  Remove the comment.

ANovokmet commented on src/project.ts:
```diff
@@ -0,0 +1,113 @@
+    }
+    
+    //TODO IDK what the property node's type is and it seems like it can really be anything. You would know more about what this can be
+    getPropertyNode(key: string): any {
```
Yeah I know... It's `string`.

ANovokmet commented on src/solution.ts:
```diff
@@ -0,0 +1,95 @@
+import SlnParser, { GlobalSectionNode, HeaderNode } from "./parser";
+import Project from "./project";
+
+class Solution {
```
export default

ANovokmet reviewed (commented):
Nice work. There is one thing missing, index.js should be rewritten into typescript too. You can move it to /src. Also, when using require for default export we use `Solution: require('./dist/solution').default`.

KolCrooks committed:
Made changes that we talked about

KolCrooks commented:
I added the changes that you mentioned. I also changed the parser to have different files. I can revert these changes if you want. The only downside to the way I changed it is that when you import the parser, you have to do `import * as SlnParser` instead of `import SlnParser`. Other than that I think that this is pretty good.

ANovokmet commented:
Yes, it looks pretty good. Thanks!
